app/Filament/Resources/UserResource/Pages/ListUsers.php

```diff
@@ -154,6 +156,14 @@ public function table(Table $table): Table
                     ->multiple()
                     ->searchable()
                     ->preload(),
+                SelectFilter::make('roles')
+                    ->label('Roles')
+                    ->options($this->getRolesOption())
+                    ->getSearchResultsUsing(fn (string $search) => Role::query()->where('name', 'like', '%' . $search . '%')->take(50)->pluck('name', 'id')->toArray())
```

Comment:
All of our searches are case-insensitive, so we will want to make sure this one is, too.

```suggestion
                    ->getSearchResultsUsing(fn (string $search) => Role::query()->where(DB::raw("LOWER('name')"), 'like', '%' . strtolower($search) . '%')->take(50)->pluck('name', 'id')->toArray())
```

Wrote this in the GitHub UI, so please test to make sure this works.